From 19bffabc0b83b754da56e43dcda8d34800e5c6f8 Mon Sep 17 00:00:00 2001
From: Kuninori Morimoto <kuninori.morimoto.gx@renesas.com>
Date: Mon, 10 Jun 2013 10:32:28 +0900
Subject: [PATCH 545/611] LTSI: ARM: shmobile: r8a7778: tidyup sound card
 detection order

Latest ASoC framework has many patches, and some of these are
controlling sound card detection order.
But, these are not back-ported yet.
This patch use quick-hack to solve this issue.

Signed-off-by: Kuninori Morimoto <kuninori.morimoto.gx@renesas.com>
---
 arch/arm/mach-shmobile/setup-r8a7778.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-shmobile/setup-r8a7778.c b/arch/arm/mach-shmobile/setup-r8a7778.c
index bf40668..cc595bb 100644
--- a/arch/arm/mach-shmobile/setup-r8a7778.c
+++ b/arch/arm/mach-shmobile/setup-r8a7778.c
@@ -253,7 +253,7 @@ void __init r8a7778_add_rsnd_devices(struct rcar_snd_info *rsnd_info,
 		rsnd_resources, ARRAY_SIZE(rsnd_resources),
 		rsnd_info, sizeof(*rsnd_info));
 
-	for (i = 0; i < card_info_nr; i++)
+	for (i = card_info_nr - 1; i >= 0; i--)
 		platform_device_register_data(
 			&platform_bus, "asoc-simple-card", i,
 			&card_info[i], sizeof(struct asoc_simple_card_info));
-- 
1.8.2

